src/deepsparse/utils/onnx.py

```diff
@@ -475,20 +482,56 @@ def truncate_onnx_embedding_model(
     return output_filepath, tmp_file
 
 
-def default_cached_outputs(model_path: str) -> List[bool]:
+def default_cached_outputs(model: Union[str, ModelProto]) -> List[bool]:
     """
+    Get a list of bools that indicate which outputs should be cached.
+    The elements that are set to True correspond to cached outputs,
+    the rest are set to False.
+
     :param model_path: Path to a model
-    :return A list of bools that indicates caching of all outputs except the first one.
+    :return A list of bools that indicate which outputs should be cached.
     """
-
-    outputs = list(onnx.load(model_path).graph.output)
+    model = (
+        onnx.load(model, load_external_data=False) if isinstance(model, str) else model
+    )
+    outputs = model.graph.output
     assert len(outputs) > 0
 
-    # Create a boolean list of every output of the
-    # model [logits, key0, value0, key1, value1, ..., keyN, valueN]
-    cached_outputs = [True for i in range(len(outputs))]
+    return [output.name.startswith(CACHE_OUTPUT_NAME) for output in outputs]
 
-    # Assume first input is logits and logits ought not to be cached
-    cached_outputs[0] = False
 
-    return cached_outputs
+def has_model_kv_cache(model: Union[str, ModelProto]) -> bool:
+    """
+    Check whether a model has a KV cache support.
+
+    :param model_path: Path to a model or a model proto.
+    :return True if the model has a KV cache support, False otherwise.
+    """
+    return bool(any(default_cached_outputs(model)))
+
+
+def overwrite_sequence_length(
+    model_path: str, sequence_length: Optional[int] = None
+) -> str:
+    """
+    Takes a path to an onnx model and enforces that it has
+    static input dimensions.
+
+    :param model_path: Path to a model.
+    :param sequence_length: The sequence length to overwrite the model with.
+    :return: Path to the model with static input dimensions.
+    """
+    from deepsparse.transformers.utils.helpers import (
+        overwrite_onnx_model_inputs_for_kv_cache_models,
+    )
+
+    onnx_file_path, _, _ = overwrite_onnx_model_inputs_for_kv_cache_models(
+        onnx_file_path=model_path, sequence_length=sequence_length
+    )
```

Comment:
Here, `sequence_length` can be `None` but `overwrite_onnx_model_inputs_for_kv_cache_models` only accepts ints. That means if `None` is passed an exception is thrown